packages/orchestrator-ui-components/src/components/WfoPageTemplate/WfoSidebar/WfoStartCreateWorkflowButtonComboBox.tsx

```diff
@@ -40,7 +48,19 @@ export const WfoStartCreateWorkflowButtonComboBox = () => {
             )
             .sort((a, b) => a.label.localeCompare(b.label)) ?? [];
 
-    const handleOptionChange = (selectedProduct: WorkflowComboBoxOption) => {
+    const handleOptionChange = async (
```

Comment:
Part of this code seems to repeat a lot. Could it be abstracted into a useCheckEngineRunning hook or something?

Reply:
Done, introduced new hook -- moved logic there, including the toast